Alignment/include/ActsAlignment/Kernel/detail/AlignmentEngine.hpp

```diff
@@ -103,7 +103,7 @@ void resetAlignmentDerivative(Acts::AlignmentToBoundMatrix& alignToBound,
 /// ingredients
 template <typename traj_t, typename parameters_t = BoundTrackParameters>
```

Comment:
I think I'll use this opportunity to add concepts here. This will then serve the purpose of documenting what kind of type we expect here.

Reply:
Not sure we want to bundle this two changes. I.e. this will delay the PR and I was hoping for an express service as the changes are quite wide and will likely catch conflicts soon.

Reply:
@paulgessinger - are you ok with leaving concept checks for next iteration? 
I believe it should only happen in Core library.